ui/src/testing/stubs/cdr-versions-api-stub.ts

```diff
@@ -1,53 +1,90 @@
+        archivalStatus: ArchivalStatus.LIVE,
+        hasMicroarrayData: true,
+        hasFitbitData: true,
+        hasWgsData: true,
+        creationTime: 0
+      },
+      {
+        name: CdrVersionsStubVariables.ALT_WORKSPACE_CDR_VERSION,
+        cdrVersionId: CdrVersionsStubVariables.ALT_WORKSPACE_CDR_VERSION_ID,
+        accessTierShortName: AccessTierShortNames.Registered,
+        archivalStatus: ArchivalStatus.LIVE,
+        hasMicroarrayData: false,
+        hasFitbitData: true,
+        hasWgsData: false,
+        creationTime: 0
+      },
+    ]},
     {
-      name: CdrVersionsStubVariables.DEFAULT_WORKSPACE_CDR_VERSION,
-      cdrVersionId: CdrVersionsStubVariables.DEFAULT_WORKSPACE_CDR_VERSION_ID,
-      accessTierShortName: AccessTierShortNames.Registered,
-      archivalStatus: ArchivalStatus.LIVE,
-      hasMicroarrayData: true,
-      hasFitbitData: true,
-      hasWgsData: true,
-      creationTime: 0
-    },
-    {
-      name: CdrVersionsStubVariables.ALT_WORKSPACE_CDR_VERSION,
-      cdrVersionId: CdrVersionsStubVariables.ALT_WORKSPACE_CDR_VERSION_ID,
-      accessTierShortName: AccessTierShortNames.Registered,
-      archivalStatus: ArchivalStatus.LIVE,
-      hasMicroarrayData: false,
-      hasFitbitData: true,
-      hasWgsData: false,
-      creationTime: 0
-    },
+      accessTierShortName: AccessTierShortNames.Controlled,
+      defaultCdrVersionId: CdrVersionsStubVariables.CONTROLLED_TIER_CDR_VERSION_ID,
+      versions: [
+        {
+          name: CdrVersionsStubVariables.CONTROLLED_TIER_CDR_VERSION,
+          cdrVersionId: CdrVersionsStubVariables.CONTROLLED_TIER_CDR_VERSION_ID,
+          accessTierShortName: AccessTierShortNames.Controlled,
+          archivalStatus: ArchivalStatus.LIVE,
+          hasMicroarrayData: true,
+          hasFitbitData: true,
+          hasWgsData: true,
+          creationTime: 0
+        }
+      ]}
   ]
 };
 
-export class CdrVersionsApiStub extends CdrVersionsApi {
-  public cdrVersions: CdrVersion[];
+const registeredCdrVersionTier = cdrVersionTiersResponse.tiers[0];
+const defaultCdrVersion = registeredCdrVersionTier.versions[0];
+const altCdrVersion = registeredCdrVersionTier.versions[1];
+
+const controlledCdrVersionTier = cdrVersionTiersResponse.tiers[1];
+const controlledCdrVersion = controlledCdrVersionTier.versions[0];
+
+class CdrVersionsApiStub extends CdrVersionsApi {
   constructor() {
     super(undefined, undefined, (..._: any[]) => { throw stubNotImplementedError; });
-    this.cdrVersions = cdrVersionListResponse.items;
   }
 
-  getCdrVersions(options?: any): Promise<CdrVersionListResponse> {
-    return new Promise<CdrVersionListResponse>(resolve => {
-      resolve({
-        ...cdrVersionListResponse,
-        items: this.cdrVersions,
-      });
+  getCdrVersionsByTier(options?: any): Promise<CdrVersionTiersResponse> {
+    return new Promise<CdrVersionTiersResponse>(resolve => {
+      resolve(cdrVersionTiersResponse);
     });
   }
 }
+
+export {
```

Comment:
Why the change to this style of export?

Reply:
Going off the advice of style-guide.md

Reply:
I didn't realize that. A quick scan of the code seems to show that this style is not heavily used, but it is used in some places.

What is your preference for coding style?

I prefer exporting where the function is defined. When I read the code I find having that information close together is useful, otherwise I would have to navigate to the bottom of the file or to see if the function is meant to be used externally. I don't recall ever being in a situation where I need to scan the file to see what all the exports are - for me, that is not the way I read code. Having the breadth of all exports for a given file in one place isn't something I am usually after, but I could grep it out of the file if needed. I think it would be harder to do the reverse operation (put the exports close).

This is a bit out of scope for this PR. But I would like to get the general feel / opinions of the AoU team around this.

Reply:
I agree that it would be good to get a team consensus for style questions like these.

I don't have a real opinion on this particular point.

Reply:
Seems like that initial style guide was checked in with minimal review, with the intent that it was capturing our "effective" codebase style at the time - so asserting this specific export style was probably an overreach, or just too localized. I also have a preference to just export with the declaration (and this is what I've generally done in practice).

I definitely support switching the style guidance if you want to propose the change (ideally we could also switch the codebase to be consistent).